### PR TITLE
pipx: add new states

### PR DESCRIPTION
**Pull request by russoz**
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

pipx

### CODE REVIEW

File: plugins/modules/pipx.py [hunk 1]
```diff
@@ -128,6 +146,13 @@
         type: bool
         default: false
         version_added: 9.4.0
+    spec_metadata:
+        description:
+            - Spec metadata file for O(state=install_all).
+            - This content of the file is usually generated with C(pipx list --json), and it can be obtained with M(community.general.pipx_info)
+              with O(ignore:include_raw=true) and obtaining the content from the RV(ignore:raw_output).
```
Comment (felixfontein) [inline plugins/modules/pipx.py:153]:
```suggestion
              with O(community.general.pipx_info#module:include_raw=true) and obtaining the content from the RV(community.general.pipx_info#module:raw_output).
```

File: plugins/modules/pipx.py [hunk 2]
```diff
@@ -26,13 +26,31 @@
...
+            - pin
+            - unpin
         default: install
         description:
             - Desired state for the application.
             - The states V(present) and V(absent) are aliases to V(install) and V(uninstall), respectively.
             - The state V(latest) is equivalent to executing the task twice, with state V(install) and then V(upgrade).
               It was added in community.general 5.5.0.
+            - The states V(install_all), V(uninject), V(upgrade-shared), V(pin) and V(unpin) are only available in C(pipx>=1.6.0),
+              make sure to have a compatible version when using this option.
```
Comment (felixfontein) [inline plugins/modules/pipx.py:53]: You should also mention that the new states were added in community.general 9.4.0.

**Review by felixfontein — COMMENTED** [review]
Sorry for the late update...

File: plugins/modules/pipx.py (hunk 2 above)
Comment (russoz) [reply plugins/modules/pipx.py]: done

File: plugins/modules/pipx.py [hunk 3]
```diff
@@ -26,13 +26,31 @@
...
+            - latest
+            - pin
+            - unpin
         default: install
         description:
             - Desired state for the application.
             - The states V(present) and V(absent) are aliases to V(install) and V(uninstall), respectively.
             - The state V(latest) is equivalent to executing the task twice, with state V(install) and then V(upgrade).
               It was added in community.general 5.5.0.
+            - The states V(install_all), V(uninject), V(upgrade-shared), V(pin) and V(unpin) are only available in C(pipx>=1.6.0),
```
Comment (felixfontein) [inline plugins/modules/pipx.py:52]:
```suggestion
            - The states V(install_all), V(uninject), V(upgrade_shared), V(pin) and V(unpin) are only available in C(pipx>=1.6.0),
```

File: tests/integration/targets/pipx/files/spec.json.license [hunk 4]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (felixfontein) [inline tests/integration/targets/pipx/files/spec.json.license:1]:
```suggestion
```

**Review by felixfontein — COMMENTED** [review]
Sorry, two more details ;) Feel free to merge once these are fixed and CI passes. Otherwise I'll do it on the weekend.

**Comment by felixfontein** [comment]
@russoz thanks for your contribution!

**Pull request merged**